libs/orderbook/src/lib/orderbook-manager.tsx

```diff
@@ -0,0 +1,77 @@
+import { AsyncRenderer } from '@vegaprotocol/ui-toolkit';
+import { Orderbook } from './orderbook';
+import { useDataProvider } from '@vegaprotocol/react-helpers';
+import { marketDepthDataProvider } from './market-depth-data-provider';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import type { MarketDepthSubscription_marketDepthUpdate } from './__generated__/MarketDepthSubscription';
+import { compact, updateCompactedData } from './orderbook-data';
+import type { OrderbookData } from './orderbook-data';
+
+interface OrderbookManagerProps {
+  marketId: string;
+  resolution: number;
+}
+
+export const OrderbookManager = ({
+  marketId,
+  resolution,
+}: OrderbookManagerProps) => {
+  const variables = useMemo(() => ({ marketId }), [marketId]);
+  const resolutionRef = useRef(resolution);
+  const lastUpdateRef = useRef(new Date().getTime());
+  const [orderbookData, setOrderbookData] = useState<OrderbookData[] | null>(
+    null
+  );
+  const dataRef = useRef<OrderbookData[] | null>(null);
+
+  // Apply updates to the table
+  const update = useCallback(
+    (delta: MarketDepthSubscription_marketDepthUpdate) => {
+      if (!dataRef.current) {
+        return false;
+      }
+      dataRef.current = updateCompactedData(
```

Comment:
chore (blocking): Its not immediately clear why we need dataRef and can't just use some react state. Could you add some comments to clarify